sc2/portconfig.py

```diff
@@ -15,11 +45,20 @@ def __str__(self):
     def as_json(self):
         return json.dumps({"shared": self.shared, "server": self.server, "players": self.players})
 
+    @classmethod
+    def contiguous_ports(cls, attempts=40):
+        """Returns a Portconfig with adjacent ports"""
+        for _ in range(attempts):
+            start = portpicker.pick_unused_port()
+            others = [start + j for j in range(1, 4)]
+            if all(portpicker.is_port_free(p) for p in others):
+                pc = cls(server_ports=[start, others[0]], player_ports=[others[1], others[2]])
+                pc._picked_ports.append(start)
+                return pc
+        else:
+            raise portpicker.NoFreePortFoundError()
+
     @classmethod
     def from_json(cls, json_data):
```

Comment:
Perhaps this could benefit if this class could be converted to a dataclass and make use of https://pypi.org/project/dataclasses-json/ (Create class from json or from dict)